### ISSUE TITLE
Error in server console

### ISSUE THREAD

**Issue by ghost** [TelepathicGrunt/RepurposedStructures-Fabric#218]
using repurposed strucutres 6.1.1-1.19
fabric api 57.0-1.19
fabric loader 0.14.8-1.19

[11:02:20 ERROR]: Registry is already frozen (requested key: ResourceKey[minecraft:worldgen/processor_list / repurposed_structures:pyramids/nether_randomizer])
[11:02:20 ERROR]: Exception loading structure piece with id minecraft:jigsaw
java.lang.IllegalStateException: Invalid pool element found
	at net.minecraft.class_3790.method_35448(class_3790.java:54) ~[server-intermediary.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_3790.<init>(class_3790.java:54) ~[server-intermediary.jar:?]
	at net.minecraft.class_6624.method_38711(class_6624.java:73) ~[server-intermediary.jar:?]
	at net.minecraft.class_3449.method_41621(class_3449.java:66) ~[server-intermediary.jar:?]
	at net.minecraft.class_2852.method_12392(class_2852.java:480) ~[server-intermediary.jar:?]
	at net.minecraft.class_2852.method_12395(class_2852.java:211) ~[server-intermediary.jar:?]
	at net.minecraft.class_3898.method_43375(class_3898.java:624) ~[server-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:157) ~[server-intermediary.jar:?]
	at net.minecraft.class_3215$class_4212.method_18859(class_3215.java:557) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:131) ~[server-intermediary.jar:?]
	at net.minecraft.class_3215$class_4212.method_16075(class_3215.java:566) ~[server-intermediary.jar:?]
	at net.minecraft.class_3215.method_19492(class_3215.java:279) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:764) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:752) ~[server-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:116) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:736) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:669) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:258) ~[server-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]

**Comment by TelepathicGrunt** [comment]
Duplicate of https://github.com/TelepathicGrunt/RepurposedStructures-Quilt/issues/199

@cpm9 please remove any outdated 1.18.2 or older RS datapack if you have any. If this is an old world you are trying to port up to 1.19, then use this datapack to try and make the chunks load again with their outdated data. https://github.com/TelepathicGrunt/RepurposedStructures-Quilt/issues/199#issuecomment-1132217839

Basically, updating old modded world to newer mc versions is typically a recipe for disaster and issues. Most likely case is the chunks are saved with the old outdated data and cannot be parsed correctly anymore. The datapack I linked above brings back the processors the chunk is looking for as dummy processors so it can stop crashing.

**Comment by ghost** [comment]
Okay if thats the case I will no longer be using this mod on my server thanks

**Comment by TelepathicGrunt** [comment]
I mean, the datapack I linked above would work as others verified it lol. But that's fine. However, I cannot promise that other structure mods or datapacks that you are using have not also refactored their processor list since 1.18.2 or older and would cause the same crash for their structures now in 1.19. it's just a bug with how vanilla reads structure data from the chunks

Hmm. Wouldn't be a bad idea for this to be reported to Fabric API so they can patch the bug.
